kernel/Utilities/GlobalIndexing.cpp

```diff
@@ -226,39 +252,15 @@ void GlobalIndexing::constructUnblocked() {
     // Construct local ordering.
     std::size_t index = 0;
     for (Base::Element *element : mesh_->getElementsList()) {
-        for (std::size_t unknown : includedUnknowns_) {
-            offsets_[unknown].elementOffsets_[element->getID()] = index;
-            index += element->getLocalNumberOfBasisFunctions(unknown);
-        }
-
-        for (Base::Face *face : element->getFacesList()) {
-            if (face->getPtrElementLeft() == element) {
+        visitElementOwnedParts(
+            element, mesh_->dimension(), [&](Base::MeshEntity *entity) {
                 for (std::size_t unknown : includedUnknowns_) {
-                    offsets_[unknown].faceOffsets_[face->getID()] = index;
-                    index += face->getLocalNumberOfBasisFunctions(unknown);
-                }
-            }
-        }
-
-        for (Base::Edge *edge : element->getEdgesList()) {
-            if (edge->getElement(0) == element) {
-                for (std::size_t unknown : includedUnknowns_) {
-                    offsets_[unknown].edgeOffsets_[edge->getID()] = index;
-                    index += edge->getLocalNumberOfBasisFunctions(unknown);
-                }
-            }
-        }
-        // Faces and nodes are the same in 1D
-        if (mesh_->dimension() > 1) {
-            for (Base::Node *node : element->getNodesList()) {
-                if (node->getElement(0) == element) {
-                    for (std::size_t unknown : includedUnknowns_) {
-                        offsets_[unknown].nodeOffsets_[node->getID()] = index;
-                        index += node->getLocalNumberOfBasisFunctions(unknown);
-                    }
+                    OffsetVisitor visitor(offsets_[unknown]);
+                    entity->visitEntity(visitor);
+                    *(visitor.entityOffset_) = index;
+                    index += entity->getLocalNumberOfBasisFunctions(unknown);
```

Comment:
I do not understand what this is doing maybe make it a function?

Reply:
I've added several lines of documentation instead, as this is quite one-of and it does not really make a nice function (as it is hardly separable from the surrounding code).